tests_integration/__tests__/__snapshots__/support-info.js.snap

```diff
@@ -676,7 +677,13 @@ exports[`CLI --support-info (stdout) 1`] = `
         \\".yaml.sed\\",
         \\".yml.mysql\\"
       ],
-      \\"filenames\\": [\\".clang-format\\", \\".clang-tidy\\", \\".gemrc\\", \\"glide.lock\\"],
+      \\"filenames\\": [
+        \\".clang-format\\",
+        \\".clang-tidy\\",
+        \\".gemrc\\",
+        \\"glide.lock\\",
+        \\"yarn.lock\\"
```

Comment:
I've updated the snapshot because the test was failing. However, we don't need to format `yarn.lock`, so I think we should ignore it by default.

Reply:
But what if someone does want to format it? Adding a `--with-yarn-lock` option similar to `--with-node-modules` sounds like an overkill... 🤔

How about simply excluding it when a directory is expanded (`getSupportedFilesGlob` in `cli/expand-patterns.js`)? Thus it's still would be possible to format if by specifying the file name explicitly: `prettier yarn.lock`, but it'll be excluded from `prettier .`.

Reply:
@thorn0 Sounds good! I'm not familiar with the file, so can you work on it?

Reply:
ok